src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -289,7 +256,10 @@ object CheckHighForm extends Pass with LazyLogging {
     }
     sinfo = c.info
     if (numTopM != 1) errors.append(new NoTopModuleException(c.main))
-    if (errors.nonEmpty) throw new PassExceptions(errors)
+    if (errors.nonEmpty) {
+      errors.append(new PassException(s"${errors.length} high form error(s) detected"))
+      throw new PassExceptions(errors)
```

Comment:
I think making errors a class makes the most sense, I'll look into it.

I'd also like an elegant (and threadsafe) way to provide commandline options that could disable "grouped" error reporting, instead giving the more specific exception so we can check it in testing. I'm thinking maybe we pass around a CircuitContext object along with the Circuit that contains the errors object. At instantiation time the errors object could be told to either relay the exception or save it to be thrown as a group later.